programmers/난이도별/level01.키패드_누르기/SSSOy.java

```diff
@@ -0,0 +1,43 @@
+class Solution {
+    
+    static int keypad[][] = {{0, 0}, {0, 1}, {0, 2}, {1, 0}, {1, 1}, {1, 2}, {2, 0}, {2, 1}, {2, 2}, {3, 0}, {3, 1}, {3, 2}};
+    static int fingers[][] = {{3, 0}, {3, 2}};
+    
+    public String solution(int[] numbers, String hand) {
+        String answer = "";
+        
+        for(int i = 0; i < numbers.length; i++) {
```

Comment:
```suggestion
int numLen = numbers.length;
        for(int i = 0; i < numLen; i++) {
```
이것도 황준일님께서 알려주신 것처럼 하면 좋을 것 같습니다. ([참고](https://github.com/Mirim-Study/Algorithm/pull/75#discussion_r484275115))

Reply:
배열의 length는 배열의 길이를 가지는 멤버변수인데, 멤버변수의 값을 가져오는 것이 실행 횟수나 복잡도? 등에 영향이 있는 지 궁금합니다. 
[참고](https://offbyone.tistory.com/192)

Reply:
> 조건에 .length .length() .size() 같은 것들이 있으면 반복문을 실행할 때 마다 해당 메소드를 실행하게됩니다.
예를들어 10000번 반복한다 치면, 10000번씩 배열의 길이를 계산해서 넘겨주는거죠!
그래서 배열이나 리스트의 길이가 변하는게 아니라면 반복문이 실행되기 전에 혹은 for문에서 미리 선언하여 사용하는 것을 권합니다.

배열의 length는 멤버변수가 아니라 메소드 아닌가요? 위의 참고링크 봐주세요

Reply:
조금더 다르게 접근하면 어떨까요?
현재 반복문에서 i는 굳이 사용할필요 없습니다.
그래서 forEach 형태로 사용하면 좋을 것 같아요!

Reply:
아하 그런 방법은 생각을 못 했네요 감사합니다!!😊

++
[참고](https://offbyone.tistory.com/192) <- 여기 링크를 보면 

_"배열은 배열의 길이를 가지는 상수 멤버변수인 length를 가집니다. "_

라고 되어 있습니다! 따라서 제 생각에는 .length를 사용할 때 멤버변수의 값을 불러오는 것이기 때문에, 
길이가 여러 번 계산되는 것은 아닌 것 같습니다. 혹시 배열의 .length에 대해서 자세히 아시는 분이 있다면
도움 주시면 감사하겠습니다ㅎㅎ!🙏